### PR TITLE
feat(mem-pool): auto-create accounts for native pCKB transfer recipients

### PR DESCRIPTION
**Pull request by zeroqn**
Filter ckb transfer log and save new recipient addresses to mem block. Create accounts after mem pool reset.

require minimal balance: 1 ckb

### CODE REVIEW

File: crates/mem-pool/src/account_creator.rs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    wallet::Wallet,
+    RollupContext,
+};
+use tracing::instrument;
+
+const META_CONTRACT_ACCOUNT_ID: u32 = RESERVED_ACCOUNT_ID;
+const ONE_CKB: u64 = 10u64.pow(8);
+pub const MIN_BALANCE: u64 = ONE_CKB;
+
+pub struct CkbTransfer;
```
Comment (magicalne) [inline crates/mem-pool/src/account_creator.rs]: For me, it's not straightforward to understand this. It's weird to see the `CkbTransfer` struct, which is empty, in  account_creteor.rs.
How about `fn filter_new_address`  only or `filter_new_created_address`?

File: crates/mem-pool/src/account_creator.rs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+impl CkbTransfer {
+    pub fn filter_new_address<'a>(
+        logs: impl IntoIterator<Item = LogItemReader<'a>>,
+        state: &impl State,
+    ) -> Option<Vec<RegistryAddress>> {
+        let do_filter = |log: LogItemReader<'_>| -> Result<Option<RegistryAddress>> {
+            if !matches!(log.service_flag().into(), GW_LOG_SUDT_TRANSFER)
+                || !matches!(log.account_id().unpack(), CKB_SUDT_ACCOUNT_ID)
```
Comment (blckngm) [inline crates/mem-pool/src/account_creator.rs]: Question: could this kind of logs be forged?

And nit: no need to use `matches!` when you can use `!=`:

```
            if (log.service_flag().into(), log.account_id().unpack())
                != (GW_LOG_SUDT_TRANSFER, CKB_SUDT_ACCOUNT_ID)
```

File: crates/mem-pool/src/mem_block.rs [hunk 3]
```diff
@@ -93,6 +96,7 @@ impl MemBlock {
         let content = MemBlockContent {
             txs: self.txs.clone(),
             withdrawals: self.withdrawals.clone(),
+            new_addresses: self.new_addresses.drain().collect(),
```
Comment (blckngm) [inline crates/mem-pool/src/mem_block.rs]: Nit: use `std::mem::take` to take `txs`, `withdrawals` and `new_addresses`.

**Review by blckngm — DISMISSED** [review]

File: crates/mem-pool/src/account_creator.rs (hunk 1 above)
Comment (zeroqn) [reply crates/mem-pool/src/account_creator.rs]: Fixed

File: crates/mem-pool/src/account_creator.rs (hunk 2 above)
Comment (zeroqn) [reply crates/mem-pool/src/account_creator.rs]: Answer: No. Only sudt contract can emit this log.

Fixed

File: crates/mem-pool/src/mem_block.rs (hunk 3 above)
Comment (zeroqn) [reply crates/mem-pool/src/mem_block.rs]: Fixed

**Review by magicalne — APPROVED** [review]

**Review by jjyr — APPROVED** [review]

**Pull request merged**